File: Eclipse Projects/webapps/TeleDirectPhp/WebContent/scripts/php/Test/TestUser.php
<?php
	require_once 'User.php';
	
	class TestUser extends PHPUnit_Framework_TestCase{
		protected $user, $username, $name, $priveleges;
		
		protected function setUp(){
			$this->username = "akash";
			$this->name = "Akash Gupta";
			$this->priveleges = "rw";
			
			$this->user = new User($this->username, $this->name, $this->priveleges);
		} 
		
		/** @test **/
		public function testGetUsername(){
			$this->assertSame($this->user->getUsername(), $this->username);
		}
		
		public function testGetName(){
			$this->assertSame($this->user->getName(), $this->name);
		}
		
		public function testGetPriveleges(){
			$this->assertSame($this->user->getPriveleges(), $this->priveleges);
		}
		
		public function testToArray(){
			$array = array(
				'username' => $this->username,
				'name' => $this->name,
				'priveleges' => $this->priveleges	
			);
			
			$this->assertSame($this->user->toArray(), $array);
			
		}
	}
?>

File: Eclipse Projects/webapps/TeleDirectJavaServer/src/com/test/teleDirect/TestUser.java
package com.test.teleDirect;

import static org.junit.Assert.*;

import org.junit.Before;
import org.junit.Test;

import com.teleDirect.User;

public class TestUser {

	private User user;
	private String username;
	private String name;
	private String priveleges;
	
	@Before
    public void setUp() {
		username = "akash";
		name = "Akash";
		priveleges = "rw";
		user = new User(username, name, priveleges);
    }
	
	@Test
	public void shouldBeAbleToGetUsername() {
		assertEquals(username, user.getUsername());
	}

	@Test
	public void shouldBeAbleToGetName() {
		assertEquals(name, user.getName());
	}
	
	@Test
	public void shouldBeAbleToGetPriveleges() {
		assertEquals(priveleges, user.getPriveleges());
	}

	@Test
	public void shouldBeAbleToSetUsername() {
		String username2 = "sagar";
		user.setUsername(username2);
		assertEquals(username2, user.getUsername());
	}
	
	@Test
	public void shouldBeAbleToSetName() {
		String name2 = "Sagar";
		user.setName(name2);
		assertEquals(name2, user.getName());
	}
	
	@Test
	public void shouldBeAbleToSetPriveleges() {
		String priveleges2 = "r";
		user.setPriveleges(priveleges2);
		assertEquals(priveleges2, user.getPriveleges());
	}

}
